### ISSUE TITLE
Deprecation warnings + session not persisting

### ISSUE THREAD

**Issue by Ghostavio** [MattMSumner/torii-fire#6]
There are several deprecation warnings when using Ember 1.13.2 and Ember Data 1.13.4 (latest versions). Also the session is not persisting after reloading the page.

**Comment by Ghostavio** [comment]
First Deprecation Warning:

```
Ember Inspector (Deprecation Trace): Adapter#find has been deprecated and renamed to `findRecord`.

        at ember$data$lib$system$store$finders$$_find (http://localhost:4200/assets/vendor.js:75651:15)
        at ember$data$lib$system$store$$Service.extend.fetchRecord (http://localhost:4200/assets/vendor.js:80012:23)
        at _fetchRecord (http://localhost:4200/assets/vendor.js:80066:53)
        at ember$data$lib$system$store$$Service.extend._flushPendingFetchForType (http://localhost:4200/assets/vendor.js:80110:11)
        at Map.forEach.cb (http://localhost:4200/assets/vendor.js:27376:20)
        at OrderedSet.forEach (http://localhost:4200/assets/vendor.js:27163:11)
        at Map.forEach (http://localhost:4200/assets/vendor.js:27384:18)
        at ember$data$lib$system$store$$Service.extend.flushAllPendingFetches (http://localhost:4200/assets/vendor.js:80055:28)
        at Queue.invoke (http://localhost:4200/assets/vendor.js:11455:16)
```

**Comment by Ghostavio** [comment]
Second Deprecation Warning:

```
Ember Inspector (Deprecation Trace): Your custom serializer uses the old version of the Serializer API, with `extract` hooks. Please upgrade your serializers to the new Serializer API using `normalizeResponse` hooks instead.

        at ember$data$lib$system$store$serializer$response$$normalizeResponseHelper (http://localhost:4200/assets/vendor.js:74034:15)
        at http://localhost:4200/assets/vendor.js:75666:25
        at Object.Backburner.run (http://localhost:4200/assets/vendor.js:10776:25)
        at ember$data$lib$system$store$$Service.extend._adapterRun (http://localhost:4200/assets/vendor.js:81352:33)
        at http://localhost:4200/assets/vendor.js:75664:22
        at tryCatch (http://localhost:4200/assets/vendor.js:65295:14)
        at invokeCallback (http://localhost:4200/assets/vendor.js:65310:15)
        at publish (http://localhost:4200/assets/vendor.js:65278:9)
        at http://localhost:4200/assets/vendor.js:42094:7
```

**Comment by Ghostavio** [comment]
Third one:

```
Ember Inspector (Deprecation Trace): Passing classes to store methods has been removed. Please pass a dasherized string instead of undefined

        at ember$data$lib$system$store$$Service.extend.adapterFor (http://localhost:4200/assets/vendor.js:81340:15)
        at push (http://localhost:4200/assets/vendor.js:98325:32)
        at superWrapper [as push] (http://localhost:4200/assets/vendor.js:32111:20)
        at ember$data$lib$system$store$serializer$response$$_pushResourceObject (http://localhost:4200/assets/vendor.js:74225:20)
        at ember$data$lib$system$store$serializer$response$$pushPayloadData (http://localhost:4200/assets/vendor.js:74190:20)
        at ember$data$lib$system$store$serializer$response$$pushPayload (http://localhost:4200/assets/vendor.js:74167:20)
        at http://localhost:4200/assets/vendor.js:75668:24
        at Object.Backburner.run (http://localhost:4200/assets/vendor.js:10776:25)
        at ember$data$lib$system$store$$Service.extend._adapterRun (http://localhost:4200/assets/vendor.js:81352:33)
```

**Comment by Ghostavio** [comment]
Fourth (kind of similar to the previous one):

```
Ember Inspector (Deprecation Trace): Passing classes to store methods has been removed. Please pass a dasherized string instead of {data: [object Object]}

        at ember$data$lib$system$store$$Service.extend.serializerFor (http://localhost:4200/assets/vendor.js:81379:15)
        at listenForChanges (http://localhost:4200/assets/vendor.js:97802:30)
        at recordWasPushed (http://localhost:4200/assets/vendor.js:97776:14)
        at push (http://localhost:4200/assets/vendor.js:98327:23)
        at superWrapper [as push] (http://localhost:4200/assets/vendor.js:32111:20)
        at ember$data$lib$system$store$serializer$response$$_pushResourceObject (http://localhost:4200/assets/vendor.js:74225:20)
        at ember$data$lib$system$store$serializer$response$$pushPayloadData (http://localhost:4200/assets/vendor.js:74190:20)
        at ember$data$lib$system$store$serializer$response$$pushPayload (http://localhost:4200/assets/vendor.js:74167:20)
        at http://localhost:4200/assets/vendor.js:75668:24
```

**Comment by Ghostavio** [comment]
And last one, that happens only after you login:

```
Ember Inspector (Deprecation Trace): The default behavior of `shouldBackgroundReloadRecord` will change in Ember Data 2.0 to always return true. If you would like to preserve the current behavior please override `shouldBackgroundReloadRecord` in your adapter:application and return false.

        at Ember.Object.extend.shouldBackgroundReloadRecord (http://localhost:4200/assets/vendor.js:68776:15)
        at ember$data$lib$system$store$$Service.extend._fetchOrResolveInternalModel (http://localhost:4200/assets/vendor.js:79969:21)
        at ember$data$lib$system$store$$Service.extend._findByInternalModel (http://localhost:4200/assets/vendor.js:79938:41)
        at ember$data$lib$system$store$$Service.extend.findRecord (http://localhost:4200/assets/vendor.js:79928:21)
        at exports.default.Ember.default.Object.extend.open (http://localhost:4200/assets/melons.js:789:22)
        at initializePromise (http://localhost:4200/assets/vendor.js:65342:7)
        at new Promise (http://localhost:4200/assets/vendor.js:66859:39)
        at open (http://localhost:4200/assets/melons.js:788:14)
        at http://localhost:4200/assets/vendor.js:100080:24
```

**Comment by Ghostavio** [comment]
And also one in the adapter to replace `lookup('store:main')` by `lookup('service:store')`

**Comment by Ghostavio** [comment]
maybe this are issues on torii itself?

**Comment by MattMSumner** [comment]
possibly, let me take a look at this tonight

**Comment by MattMSumner** [comment]
I've opened a PR to update everything but I still need to test it out on a new app to see if the warnings get removed. https://github.com/MattMSumner/torii-fire/pull/7/files

**Comment by MattMSumner** [comment]
So I've gotten rid of most of the deprecations but there's a few I still have to squish. Gonna tackle the rest over the weekend. Thanks for reporting this!

**Comment by Ghostavio** [comment]
no problem, thanks for the quick response!

**Comment by MattMSumner** [comment]
@Ghostavio I'm actually trying to get torii-fire merged into [emberfire](https://github.com/firebase/emberfire) https://github.com/firebase/emberfire/pull/256

If this gets merged that should handle most of the depreciation warnings. If there are any more I'll start opening PRs against [emberfire](https://github.com/firebase/emberfire) to get them squashed.

Thanks!

**Comment by Ghostavio** [comment]
nice!

**Comment by MattMSumner** [comment]
@Ghostavio heads up this got merged!

**Comment by Ghostavio** [comment]
amazing :D 
I'll close this issue
